test/test-project/src/test/editor/editorView-test.ts

```diff
@@ -149,4 +162,81 @@ describe('EditorView', () => {
             expect(titles).deep.equals(allTitles);
         });
     });
-});
\ No newline at end of file
+});
+
+async function hasTab(title: string): Promise<boolean> {
+    const view = new EditorView();
+    const tabs = await view.getOpenTabs();
+
+    for (const tab of tabs) {
+        try {
+            if (await tab.getTitle() === title) {
+                return true;
+            }
+        } catch (e) {
+            if (e instanceof error.StaleElementReferenceError) {
+                continue;
+            }
+            throw e;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Create a new untitled file.
+ * @param expectedName expected file name; it must be in format Untitled-${x}.
+ * @param timeout time in ms, after this time windows the function is considered to be failed
+ * @param interval driver.actions() interval in ms
+ */
+async function newFile(expectedName: string, timeout: number = 10000, interval: number = 2000): Promise<void> {
```

Comment:
I think that it is quite too much complex workaround ale again it is pity to have this kind of logic directly in test class...
the new file creation could be done easier in first iteration..

but I like this kind of solid and robust approach.. we should discuss how to reuse this code and add another like that one..

AI: discuss some kind of experimental FileSystem components which could be shipped as part of extester

Reply:
Extester does not have any click/sendKeys limiters which would be nice to have and make testing more straightforward.